### ISSUE TITLE
Initial chart show is really slow

### ISSUE THREAD

**Issue by bcronje** [amcharts/amcharts4#723]
I find the time it takes from chart creation to the first time it is shown on the screen to be very slow even on basic charts. Have a look at [https://codepen.io/anon/pen/qLROKr](https://codepen.io/anon/pen/qLROKr) 

This basic example takes anywhere from 500ms up to 900ms just to render the initial chart, which seems super slow. 

Time in the above example is measured with:
```
console.time("chart");
var chart = am4core.create("chartdiv", am4charts.XYChart);
chart.events.on("shown", () => {
  console.timeEnd("chart");
});
```

Any idea what can be done to improve the initial render?

**Comment by zeroin** [comment]
What would really speed things up is if you'd change XYChartScrollbar to a regular Scrollbar, but of course, it's not the same. We are still working on improving performance.

**Comment by bcronje** [comment]
Maybe this could be of interest to you. I've update [https://codepen.io/anon/pen/qLROKr](https://codepen.io/anon/pen/qLROKr) to track the time between stages of the chart creation process using the following:

```
console.time("chart");
var chart = am4core.create("chartdiv", am4charts.XYChart);
chart.events.onAll((ev) => {
  if (ev !== "propertychanged") {
    console.log(ev);
    console.timeEnd("chart");
    console.time("chart");
  }
});
```

This gives some insight into which stages takes up most time. Output on my PC is:

```
transitionended - 0.4ms
hidden - 0.4ms
beforedatavalidated 2.3ms
transformed - 104.5ms
transformed - 0.5ms
sizechanged - 81.1ms
beforedatavalidated - 20.2ms
datarangechanged - 54.1ms
datarangechanged - 0.3ms
beforevalidated - 16.3ms
datavalidated - 204.2ms
dataitemsvalidated - 0.3ms
validated - 7.7ms
visibilitychanged - 0.6ms
shown - 0.6ms
```

So the big time spenders seems to be:
From beforedatavalidated -> first transformed - 104.5ms
From transformed -> sizechanged - 81.1ms
From beforevalidated -> datavalidated 204.2ms

What exactly happens during that expensive data validation phase?

**Comment by zeroin** [comment]
Thank you for your insights. I totally understand the problem and we here doing our best to make the initial rendering faster. During this data validation process most of the physical objects are created so that's what makes it slow. But I promise to make it faster, we just need more time :)

**Comment by bcronje** [comment]
Thank you for the feedback @zeroin and happy to hear you guys are working on improving perf. I'll go ahead and close this issue.

**Comment by zeroin** [comment]
Just wanted to let you know that we released v 4.0.17, which should perform a lot better, especially for charts with a lot of data.
